HerPublicWebsite/Startup.cs

```diff
@@ -6,309 +6,297 @@
-            services.AddScoped<IGuidService, GuidService>();
-            services.AddScoped<QuestionnaireService>();
-            services.AddScoped<QuestionnaireUpdater>();
-            services.AddScoped<IQuestionFlowService, QuestionFlowService>();
-            services.AddScoped<IUnsubmittedReferralRequestsService, UnsubmittedReferralRequestsService>();
-            services.AddScoped<IWorkingDayHelperService, WorkingDayHelperService>();
-            services.AddScoped<IDateHelper, DateHelper>();
-            services.AddScoped<IReferralFilterService, ReferralFilterService>();
-            services.AddScoped<ISessionRecorderService, SessionRecorderService>();
-            services.AddScoped<IDateTimeProvider, DateTimeProvider>();
-
-            services.AddMemoryCache();
-            services.AddSingleton<StaticAssetsVersioningService>();
-            // This allows encrypted cookies to be understood across multiple web server instances
-            services.AddDataProtection().PersistKeysToDbContext<HerDbContext>();
-
-            ConfigureReferralFollowUpNotificationService(services);
-            ConfigureS3Client(services);
-            ConfigureS3FileWriter(services);
-            ConfigureEpcApi(services);
-            ConfigureOsPlacesApi(services);
-            ConfigureGovUkNotify(services);
-            ConfigureCookieService(services);
-            ConfigureDatabaseContext(services);
-            ConfigureGoogleAnalyticsService(services);
-
-            if (!webHostEnvironment.IsDevelopment() && !webHostEnvironment.IsProduction())
-            {
-                services.Configure<BasicAuthMiddlewareConfiguration>(
-                    configuration.GetSection(BasicAuthMiddlewareConfiguration.ConfigSection));
-            }
+    public Startup(IConfiguration configuration, IWebHostEnvironment webHostEnvironment)
+    {
+        this.configuration = configuration;
+        this.webHostEnvironment = webHostEnvironment;
+    }
 
-            services.AddControllersWithViews(options =>
+    // This method gets called by the runtime. Use this method to add services to the container.
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Add Hangfire services.
+        services.AddHangfire(config => config
+            .SetDataCompatibilityLevel(CompatibilityLevel.Version_180)
+            .UseSimpleAssemblyNameTypeSerializer()
+            .UseRecommendedSerializerSettings()
+            .UsePostgreSqlStorage(configuration.GetConnectionString("PostgreSQLConnection")));
+
+        // Add the Hangfire processing server as IHostedService
+        services.AddHangfireServer();
+
+        services.AddScoped<ICsvFileCreator, CsvFileCreator>();
+        services.AddScoped<IDataAccessProvider, DataAccessProvider>();
+        services.AddScoped<IEligiblePostcodeService, EligiblePostcodeService>();
+        services.AddSingleton<IEligiblePostcodeListService, EligiblePostcodeListService>();

```

Comment:
functional change is this line

Reply:
Can you move to the below section with the other singleton please? Easier to miss it here.

Reply:
have moved